sdk/storage/azblob/zt_examples_test.go

```diff
@@ -124,15 +118,325 @@ func Example() {
+		panic("AZURE_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	handleError(err)
+
+	serviceClient, err := azblob.NewClient(serviceURL, cred, nil)
+	handleError(err)
+
+	pager := serviceClient.NewListContainersPager(&azblob.ListContainersOptions{
+		Include: azblob.ListContainersDetail{Metadata: true, Deleted: true},
+	})
+
+	for pager.More() {
+		resp, err := pager.NextPage(ctx)
+		handleError(err) // if err is not nil, break the loop.
+		for _, _container := range resp.ContainerItems {
+			fmt.Printf("%v", _container)
+		}
+	}
+}
+
+func Example_client_UploadFile() {
+	// Set up file to upload
+	fileSize := 8 * 1024 * 1024
+	fileName := "test_upload_file.txt"
+	fileData := make([]byte, fileSize)
+	err := os.WriteFile(fileName, fileData, 0666)
+	handleError(err)
+
+	// Open the file to upload
+	fileHandler, err := os.Open(fileName)
+	handleError(err)
+
+	// close the file after it is no longer required.
+	defer func(file *os.File) {
+		err = file.Close()
+		handleError(err)
+	}(fileHandler)
+
+	// delete the local file if required.
+	defer func(name string) {
+		err = os.Remove(name)
+		handleError(err)
+	}(fileName)
+
+	accountName, ok := os.LookupEnv("AZURE_STORAGE_ACCOUNT_NAME")
+	if !ok {
+		panic("AZURE_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	handleError(err)
+
+	serviceClient, err := azblob.NewClient(serviceURL, cred, nil)
+	handleError(err)
+
+	// Upload the file to a block blob
+	containerName := "testcontainer"
+	blobName := "virtual/dir/path/" + fileName
+	blockSize := 1024
+	parallelism := 3
+	_, err = serviceClient.UploadFile(context.TODO(), containerName, blobName, fileHandler,
+		&azblob.UploadFileOptions{
+			BlockSize:   int64(blockSize),
+			Parallelism: uint16(parallelism),
+			// If Progress is non-nil, this function is called periodically as bytes are uploaded.
+			Progress: func(bytesTransferred int64) {
+				fmt.Println(bytesTransferred)
+			},
+		})
+	handleError(err)
+}
+
+func Example_client_DownloadFile() {
+	// Set up file to download the blob to
+	destFileName := "test_download_file.txt"
+	destFile, err := os.Create(destFileName)
+	handleError(err)
+	defer func(destFile *os.File) {
+		err = destFile.Close()
+		handleError(err)
+	}(destFile)
+
+	accountName, ok := os.LookupEnv("AZURE_STORAGE_ACCOUNT_NAME")
+	if !ok {
+		panic("AZURE_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	handleError(err)
+
+	serviceClient, err := azblob.NewClient(serviceURL, cred, nil)
+	handleError(err)
+
+	// Perform download
+	containerName := "testcontainer"
+	blobName := "virtual/dir/path/" + destFileName
+	blockSize := 1024
```

Comment:
Why don't we kill these variables and just put the values directly on lines 313-316. Do the same for the upload file example above. Keep the examples short. Also, don't explicitly initialize any variables/fields to 0 like doenloadOffset & downloadCount.

Reply:
I just wanted to increase the readibility of the examples by writing exactly what each parameter means. But yeah we can remove the variables to reduce the clutter and present small examples.